app/controllers/landing_pages/landing.rb

```diff
@@ -2,7 +2,7 @@
 class LandingPages::InvalidAccess < StandardError; end
 class LandingPages::InvalidParameters < StandardError; end
 
-class LandingPages::LandingController < ::ActionController::Base
+class LandingPages::LandingController < ApplicationController
```

Comment:
@cabidop Inheriting the application controller here means inheriting quite a lot of logic. Could you walk me through why you're doing this? Perhaps this is to make the new "inline" pages you've added? Does it work for non-inline pages though?

Reply:
Not directly related to the inline pages (although it simplified quite a bit the implementation), but checking the [`ActionController::Base` documentation](https://api.rubyonrails.org/classes/ActionController/Base.html) seems that by default the only controller that inherits it is the `ApplicationController`, which seems to also be the approach for the default and official Discourse plugins I checked out (e.g. [`chat`](https://github.com/discourse/discourse/blob/main/plugins/chat/app/controllers/chat/base_controller.rb), [`poll`](https://github.com/discourse/discourse/blob/main/plugins/poll/app/controllers/polls_controller.rb), [`styleguide`](https://github.com/discourse/discourse/blob/main/plugins/styleguide/app/controllers/styleguide/styleguide_controller.rb), [`calendar`](https://github.com/discourse/discourse-calendar/blob/main/app/controllers/discourse_post_event/discourse_post_event_controller.rb), [`cakeday`](https://github.com/discourse/discourse-cakeday/blob/main/app/controllers/discourse_cakeday/cakeday_controller.rb), etc).

On the other hand, I have not seen anything specifically discouraging to inherit `ActionController::Base` directly so let me know your thoughts on this.

Reply:
Yes, typically you would inherit the ApplicationController because you're serving content within Discourse and you want all the logic included in the ApplicationController (e.g. all it's before_actions). However in this case, sometimes we don't want all of that logic running because a landing page will not be running inside of the main Discourse app. If there wasn't a good reason to make this change then best not to make it.